test/support.ts

```
@@ -0,0 +1,350 @@
+  const config = { ...Config[getTestDialect()] };
+  delete config.pool;
+
+  return config;
+}
+
+export function getSequelizeInstance(db: string, user: string, pass: string, options?: Options) {
+  options = options || {};
+  options.dialect = options.dialect || getTestDialect();
+
+  return new Sequelize(db, user, pass, options);
+}
+
+export async function clearDatabase(sequelize: Sequelize) {
+  const qi = sequelize.getQueryInterface();
+  await qi.dropAllTables();
+  sequelize.modelManager.models = [];
+  sequelize.models = {};
+
+  if (qi.dropAllEnums) {
+    await qi.dropAllEnums();
+  }
+
+  await dropTestSchemas(sequelize);
+}
+
+export async function dropTestSchemas(sequelize: Sequelize) {
+  const queryInterface = sequelize.getQueryInterface();
+
+  if (!queryInterface.queryGenerator._dialect.supports.schemas) {
+    await sequelize.drop({});
+
+    return;
+  }
+
+  const schemas = await sequelize.showAllSchemas();
+  const schemasPromise = [];
+  for (const schema of schemas) {
+    // @ts-expect-error
+    const schemaName = schema.name ? schema.name : schema;
+    if (schemaName !== sequelize.config.database) {
+      schemasPromise.push(sequelize.dropSchema(schemaName));
+    }
+  }
+
+  await Promise.all(schemasPromise.map(async p => p.catch((error: unknown) => error)));
+}
+
+export function getSupportedDialects() {
+  return fs.readdirSync(path.join(distDir, 'dialects'))
+    .filter(file => !file.includes('.js') && !file.includes('abstract'));
+}
+
+// TODO: type once QueryGenerator has been migrated to TS
+export function getAbstractQueryGenerator(sequelize: Sequelize): unknown {
+  class ModdedQueryGenerator extends AbstractQueryGenerator {
+    quoteIdentifier(x: string): string {
+      return x;
+    }
+  }
+
+  // @ts-expect-error
+  return new ModdedQueryGenerator({ sequelize, _dialect: sequelize.dialect });
+}
+
+export function getTestDialect(): Dialect {
+  let envDialect = process.env.DIALECT || 'mysql';
```

Why would we default to mysql here? I know it's in the original file, but seems odd to me

---

Sounds weird to me too. I think we should throw if DIALECT is not set